xarray/core/computation.py

```diff
@@ -645,15 +645,32 @@ def apply_variable_ufunc(
             def func(*arrays):
                 import dask.array as da
 
-                res = da.apply_gufunc(
+                gufunc = functools.partial(
+                    da.apply_gufunc,
                     numpy_func,
                     signature.to_gufunc_string(exclude_dims),
                     *arrays,
                     vectorize=vectorize,
                     output_dtypes=output_dtypes,
-                    **dask_gufunc_kwargs,
                 )
 
+                try:
+                    res = gufunc(**dask_gufunc_kwargs)
+                except ValueError as exc:
+                    if "with different chunksize present" in str(exc):
```

Comment:
This is checking for exact wording of an error raised by dask for chunked non-core dimensions. That seems not-very-dependable. Can we instead check for chunked core dimensions and `dask_gufunc_kwargs["allow_rechunk"] is True` and raise an error on our own? I think we could copy the code and error over from before your previous PR, and modify it slightly.

Reply:
Now you see me totally confused :grinning:
If we set `allow_rechunk=True` it will be rechunked no matter if core or non-core dimension mismatch. If we then check for core dimension chunksize > 1 (as in the previous version) and raise, how should the users pass this without rechunking themselves? 

I'll try to come up with a proposal which doesn't rely on raised dask errors.

Reply:
we could do check for core dimension chunksize > 1 (as in the previous version)  AND `not dask_gufunc_kwargs.get("allow_rechunk")`